v7/notebook_shortcode/notebook_shortcode.py

```diff
@@ -52,8 +52,8 @@ def render_notebook(self, filename, site=None, data=None, lang=None, post=None):
         export_html = HTMLExporter(config=c)
         (notebook_raw, _) = export_html.from_filename(filename)
 
-        # The raw HTML contains garbage (scripts and styles), we can’t leave it in
+        # The raw HTML contains garbage (scripts and styles). Extract only div id=notebook-container and children
```

Comment:
Seems like we could do something like the following to remove the ids (I'm not too familiar with lxml, so there might be a cleaner way iterate over the notebook elements).

```
notebook_el = notebook_html.xpath('//*[@id="notebook"]')[0]
notebook_el.attrib.pop('id')
notebook_container_el = notebook_html.xpath('//*[@id="notebook-container"]')[0]
notebook_container_el.attrib.pop('id')

notebook_code = lxml.html.tostring(notebook_el, encoding='unicode')
```
I did a quick pass over the ipython css and nikola doesn't do any selection by id, so this appears to work.

At any rate, the merged version also works (at least in my tests), despite not being valid HTML, so we can leave this for the future.